x-pack/plugins/endpoint/public/applications/endpoint/view/alerts/index.tsx

```diff
@@ -84,6 +99,30 @@ export const AlertIndex = memo(() => {
   );
 
   const [visibleColumns, setVisibleColumns] = useState(() => columns.map(({ id }) => id));
+  const formatter = new Intl.DateTimeFormat(i18n.getLocale(), {
+    year: 'numeric',
+    month: '2-digit',
+    day: '2-digit',
+    hour: '2-digit',
+    minute: '2-digit',
+    second: '2-digit',
+  });
+
+  const handleAlertClick = useCallback(
+    (event: React.MouseEvent<HTMLElement>) => {
+      if (event.target instanceof HTMLElement) {
```

Comment:
do we have to do this because of typescript?

Reply:
this actually isn't necessary. The `event.target` could be any DOM node that is a descendant of `event.currentTarget` aka the thing that the event listener is on. Any DOM node isn't necessarily an HTMLElement. DOM nodes don't all support `alertId`. 

In this case, I could just refer to `event.currentTarget`. The (synthetic) event handler is registered to an html element (the one w/ the data-* attribute) and so referring to `event.currentTarget` will work w/o the guard.

```tsx
  const handleAlertClick = useMemo(() => {
    return (event: React.MouseEvent<HTMLElement>) => {
      const alertId: string | undefined = event.currentTarget.dataset.alertId;
      if (alertId !== undefined) {
        history.push(urlWithSelectedAlert(alertId));
      }
    };
  }, [history, urlWithSelectedAlert]);
```

Reply:
okie good to know